Mathlib/ModelTheory/Semantics.lean

```diff
@@ -830,18 +830,18 @@ theorem realize_toFormula (φ : L.BoundedFormula α n) (v : α ⊕ (Fin n) → M
     · exact Fin.elim0 x
 
 @[simp]
-theorem realize_iSup (s : Finset β) (f : β → L.BoundedFormula α n)
+theorem realize_iSup [Finite β](f : β → L.BoundedFormula α n)
```

Comment:
```suggestion
theorem realize_iSup [Finite β] (f : β → L.BoundedFormula α n)
```

Reply:
This change (and the other ones where you drop the `Finset` argument) I'm less convinced by. The diff seems to become more complicated.
But changing `Fintype` to `Finite` in `genericPolyMapSurjOnOfInjOn` makes sense to me.
Would just that change already solve many of your `Fintype.ofFinite` problems?

Reply:
The diff is more complicated because the definitions of ìInf` become more complicated, but using them becomes simpler. The file Ax-Grothendieck file got a bit simpler because of this change.

The main motivation for this was #20180 . I was working with formulas `Formula (α ⊕ β)` whose semantics was the graph of a function `(α → M) → (β → M)`. `β` has to be finite for this to work.  To define a formula whose semantics is `f(x) = g(x)`, you then need an iInf over all elements of `β`.

I think `iInf` is probably a bit different from `Finset.sum` because it's basically just a forall, so I'm reasoning about it in the same way I would if `β` was an infinite type. I don't think people will often be reasoning about it by induction or anything like that. Although my experience of applied model theory might be different from other peoples. Ax-Grothendieck is an infinite collection of formulas, but the continuum hypthothesis is just one formula, so the way we use the library will be quite different..